test/Makefile

```diff
@@ -26,6 +26,7 @@ all: build
 
 build:
 	@$(ECHO_GINKGO)$@
+	GOOS=darwin $(GINKGO) build -tags=integration_tests && rm -f test.test
```

Comment:
Can you move this to a separate target (`build-darwin`)? Why `-f`?

Reply:
Yes I can
> why -f

just a habit, will change it

Reply:
@aditighag just implemented the change you asked. That's all you wanted for now right ? Or should the GitHub action you talked about be part of this PR ?